client/src/components/Chat.jsx

```diff
@@ -311,6 +345,155 @@ const Chat = () => {
 		}
 	}, [sortedMessages]);
 
+	const importKey = async (arrayBuffer) => {
+		const storedPublicKey = localStorage.getItem('importPublicKey' + app.currentChatId);
+		const publicKeyArray = new Uint8Array(JSON.parse(storedPublicKey));
+
+		const importedPublicKey = await crypto.subtle.importKey(
+			'spki',
+			storedPublicKey ? publicKeyArray : arrayBuffer,
+			{
+				name: 'RSA-OAEP',
+				hash: { name: 'SHA-256' }
+			},
+			true,
+			['encrypt']
+			)
+				console.log(importedPublicKey);
+				setImportedPublicKey(importedPublicKey);
+				if(!storedPublicKey){
+				const exportedPublicKey = await crypto.subtle.exportKey(
+					'spki',
+					importedPublicKey
+				  );
+				  const publicKeyArray = new Uint8Array(exportedPublicKey);
+				  localStorage.setItem("importPublicKey" + app.currentChatId , JSON.stringify(Array.from(publicKeyArray)));
+				}
+			
+	}
+	const convertArrayBufferToPem = (arrayBuffer, type) => {
+		const buffer = new Uint8Array(arrayBuffer);
+		const base64String = btoa(String.fromCharCode.apply(null, buffer));
+		return `-----BEGIN ${type}-----\n${base64String}\n-----END ${type}-----`;
+	  };
+
+	  function pemToArrayBuffer(pemString) {
+		const base64String = pemString && pemString
+		  .replace(/-----BEGIN (.+)-----/, '')
+		  .replace(/-----END (.+)-----/, '')
+		  .replace(/\s/g, '');
+	  
+		const binaryString = atob(base64String);
+		const byteArray = new Uint8Array(binaryString.length);
+		for (let i = 0; i < binaryString.length; i++) {
+		  byteArray[i] = binaryString.charCodeAt(i);
+		}
+	  
+		return byteArray.buffer;
+	  }
```

Comment:
same thing here, so basically every function that doesn't affect state like setState, should be in chatHelper.

Another important thing is adding comments everywhere in your code, it's part of the project structure not perfectly done yet, but please add comments everywhere